pkg/tekton/syntax/pipeline_test.go

```diff
@@ -1235,6 +1235,84 @@ func TestParseJenkinsfileYaml(t *testing.T) {
 					sh.StructureStagePrevious("A Working Stage")),
 			),
 		},
+		{
+			name: "sidecars",
+			expected: sh.ParsedPipeline(
+				sh.PipelineOptions(
+					sh.PipelineSidecar(&corev1.Container{
+						Name:  "top-level-sidecar",
+						Image: "top-level/sidecar:tag",
+					}),
+				),
+				sh.PipelineAgent("some-image"),
+				sh.PipelineStage("A Working Stage",
+					sh.StageOptions(
+						sh.StageSidecar(&corev1.Container{
+							Name:  "stage-level-sidecar",
+							Image: "stage-level/sidecar:latest",
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "shared-volume",
+									MountPath: "/shared",
+								},
+							},
+						}),
+						sh.StageVolume(&corev1.Volume{
+							Name: "shared-volume",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						}),
+						sh.StageContainerOptions(
+							sh.ContainerVolumeMount("shared-volume", "/mnt/shared"),
+						),
+					),
+					sh.StageStep(
+						sh.StepCmd("echo"),
+						sh.StepArg("hello"), sh.StepArg("world"),
+						sh.StepName("A Step With Spaces And Such"),
+					),
+				),
+			),
+			pipeline: tb.Pipeline("somepipeline-1", "jx", tb.PipelineSpec(
```

Comment:
fyi, now that #7487 has merged, this'll need to change to `tb.Pipeline("somepipeline-1", tb.PipelineNamespace("jx"), ...`, and `tb.Task(...)` will need to change to `tb.Task("somepipeline-a-working-stage-1", tb.TaskNamespace("jx"), ...`.

Reply:
thanks, I rebased and fixed it